deployment/common/changeset/transfer_ownership.go

```diff
@@ -2,70 +2,135 @@ package changeset
 
 import (
 	"fmt"
+	"math/big"
+	"time"
 
 	"github.com/ethereum/go-ethereum/accounts/abi/bind"
 	"github.com/ethereum/go-ethereum/common"
 	gethtypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/smartcontractkit/ccip-owner-contracts/pkg/proposal/mcms"
+	"github.com/smartcontractkit/ccip-owner-contracts/pkg/proposal/timelock"
 
 	"github.com/smartcontractkit/chainlink/deployment"
+	"github.com/smartcontractkit/chainlink/deployment/common/proposalutils"
+	"github.com/smartcontractkit/chainlink/deployment/common/types"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/shared/generated/burn_mint_erc677"
 )
 
-type OwnershipTransferrer interface {
-	TransferOwnership(opts *bind.TransactOpts, newOwner common.Address) (*gethtypes.Transaction, error)
-	Owner(opts *bind.CallOpts) (common.Address, error)
+type TransferToMCMSWithTimelockConfig struct {
+	TransfersByChain map[uint64][]common.Address
+	// MinDelay is for the accept ownership proposal
+	MinDelay time.Duration
 }
 
-type TransferOwnershipConfig struct {
-	// OwnersPerChain is a mapping from chain selector to the owner's contract address on that chain.
-	OwnersPerChain map[uint64]common.Address
+type Ownable interface {
+	Owner(opts *bind.CallOpts) (common.Address, error)
+	TransferOwnership(opts *bind.TransactOpts, newOwner common.Address) (*gethtypes.Transaction, error)
+	AcceptOwnership(opts *bind.TransactOpts) (*gethtypes.Transaction, error)
+	Address() common.Address
+}
 
-	// Contracts is a mapping from chain selector to the ownership transferrers on that chain.
-	Contracts map[uint64][]OwnershipTransferrer
+func loadOwnableContract(addr common.Address, client bind.ContractBackend) (common.Address, Ownable, error) {
+	// Just using the ownership interface from here.
+	c, err := burn_mint_erc677.NewBurnMintERC677(addr, client)
+	if err != nil {
+		return common.Address{}, nil, fmt.Errorf("failed to create contract: %v", err)
+	}
+	owner, err := c.Owner(nil)
+	if err != nil {
+		return common.Address{}, nil, fmt.Errorf("failed to get owner of contract: %v", err)
+	}
+	return owner, c, nil
 }
 
-func (t TransferOwnershipConfig) Validate() error {
-	// check that we have owners for the chains in the Contracts field.
-	for chainSelector := range t.Contracts {
-		if _, ok := t.OwnersPerChain[chainSelector]; !ok {
-			return fmt.Errorf("missing owners for chain %d", chainSelector)
+func (t TransferToMCMSWithTimelockConfig) Validate(e deployment.Environment) error {
+	for chainSelector, contracts := range t.TransfersByChain {
+		for _, contract := range contracts {
+			// Cannot transfer to an unknown address.
```

Comment:
I think you mean "cannot transfer ownership of an unknown address"? Meaning unknown to the address book / CLD state?